### PR TITLE
[Fix-3153] Alert information cannot be sent in cancelling state

### LINKED ISSUE #3153: [Bug] Alert information cannot be sent in cancelling state
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### Java Version

1.8

### Scala Version

2.12.x

### StreamPark Version

2.1.1

### Flink Version

1.17.1

### deploy mode

yarn-perjob

### What happened

Alert information cannot be sent in cancelling state.

### Error Exception

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
fix #3153

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java [hunk 1]
```diff
@@ -505,6 +505,8 @@ and the status is not obtained this time (flink rest server is closed),
         log.error(
             "FlinkRESTAPIWatcher query previous state was canceling and stopFrom NotFound,savePoint expired!");
         savePointService.expire(application.getId());
+        FlinkAppState flinkAppState = FlinkAppState.of(application.getState());
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java]: Hello, thanks for your contribution. We have checked and found that the job status of the assignment is CANCELED, and it was canceled through StreamaPark. We consider CANCELED to be a normal stoppage status (manually triggered by the user). Our current policy is not to issue any alerts for statuses such as CANCELED and SUCCESS. Only when the assignment FAILED，LOST，CANCELED (not stopped through Streamapark) will an alert be triggered.

**Comment by lordk911** [comment]
When running `yarn application -kill`, the 'Run Status' also displays as 'CANCELED'. If the application is mistakenly deleted, it will cause trouble. So if a Flink application is lost in the resource queue, should we give users an option to issue an alert? Is that better?

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java (hunk 1 above)
Comment (SbloodyS) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java]: Sorry for late reply. I've been quite busy these days. Will resubmit the code in the next few days...

**Comment by wolfboys** [comment]
> When running `yarn application -kill`, the 'Run Status' also displays as 'CANCELED'. If the application is mistakenly deleted, it will cause trouble. So if a Flink application is lost in the resource queue, should we give users an option to issue an alert? Is that better?

So, flink app on yarn mode, when the flink app status is "cancel"(the app status from Flink metrics), we need to further retrieve the app status from YARN. If the status in YARN is "kill", then we will send an alert.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java (hunk 1 above)
Comment (wolfboys) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java]: > Sorry for late reply. I've been quite busy these days. Will resubmit the code in the next few days...

I look forward to your driving

**Comment by lordk911** [comment]
> So, flink app on yarn mode, when the flink app status is "cancel"(the app status from Flink metrics), we need to further retrieve the app status from YARN.

thanks, retrieve the app status from YARN again will work out.

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java (hunk 1 above)
Comment (SbloodyS) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/task/FlinkRESTAPIWatcher.java]: Done. PTAL @wolfboys

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**